smrf/distribute/wind/wind_ninja.py

```diff
@@ -0,0 +1,285 @@
+        fp_vel = self.wind_ninja_path(t, 'vel')
+
+        # get wind ninja topo stats
+        ts2 = utils.get_asc_stats(fp_vel)
+        self.windninja_x = ts2['x'][:]
+        self.windninja_y = ts2['y'][:]
+
+        XW, YW = np.meshgrid(self.windninja_x, self.windninja_y)
+        self.wn_mx = XW.flatten()
+        self.wn_my = YW.flatten()
+
+        xy = np.zeros([XW.shape[0]*XW.shape[1], 2])
+        xy[:, 1] = self.wn_my
+        xy[:, 0] = self.wn_mx
+        uv = np.zeros([self.X.shape[0]*self.X.shape[1], 2])
+        uv[:, 1] = self.Y.flatten()
+        uv[:, 0] = self.X.flatten()
+
+        self.vtx, self.wts = utils.interp_weights(xy, uv, d=2)
+
+        self.init_interp = False
+
+    def distribute(self, data_speed, data_direction):
+        """Distribute the wind for the model
+
+        Arguments:
+            data_speed {DataFrame} -- wind speed data frame
+            data_direction {DataFrame} -- wind direction data frame
+        """
+
+        t = data_speed.name
+
+        if self.init_interp:
+            self.initialize_interp(t)
+
+        wind_speed, wind_direction = self.convert_wind_ninja(t)
+        self.wind_speed = wind_speed
+        self.wind_direction = wind_direction
+
+    def convert_wind_ninja(self, t):
+        """
+        Convert the WindNinja ascii grids back to the SMRF grids and into the
+        SMRF data streamself.
+
+        Args:
+            t: datetime of timestep
+
+        Returns:
+            ws: wind speed numpy array
+            wd: wind direction numpy array
+
+        """
+
+        # get the ascii files that need converted
+        fp_vel = self.wind_ninja_path(t, 'vel')
+        data_vel = np.loadtxt(fp_vel, skiprows=6)
+        data_vel_int = data_vel.flatten()
+
+        # interpolate to the SMRF grid from the WindNinja grid
+        g_vel = utils.grid_interpolate(
+            data_vel_int, self.vtx,
+            self.wts, self.X.shape)
+
+        # There will be NaN's around the edge, hanlde those first
+        if self.model_dxdy != self.wind_ninja_dxy:
+            self._logger.debug('Wind speed from WindNinja has NaN, filling')
+
+            g_vel = self.fill_data(g_vel)
+
+        # log law scale
+        g_vel = g_vel * self.ln_wind_scale
+
+        # wind direction from angle, split into u,v components then interpolate
+        fp_ang = self.wind_ninja_path(t, 'ang')
+        data_ang = np.loadtxt(fp_ang, skiprows=6)
+
+        u = np.sin(data_ang * np.pi / 180)
+        v = np.cos(data_ang * np.pi / 180)
+
+        ui = utils.grid_interpolate(
+            u.flatten(), self.vtx, self.wts, self.X.shape)
+        vi = utils.grid_interpolate(
+            v.flatten(), self.vtx, self.wts, self.X.shape)
+
+        uf = self.fill_data(ui)
+        vf = self.fill_data(vi)
+
+        g_ang = np.arctan2(uf, vf) * 180 / np.pi
+        g_ang[g_ang < 0] = g_ang[g_ang < 0] + 360
+
+        return g_vel, g_ang
+
+    def fill_data(self, g_vel):
```

Comment:
@etrujil @Hedrick-ARS @robertson-mark this is the fix for the interpolation, it will do the left side first then the bottom. It's linear 1D interpolation along each axis. If there are NaN's that aren't on the side then it will error out.